workspace/user_test.go

```diff
@@ -186,18 +188,58 @@ func TestUpdateRole(t *testing.T) {
 		}
 	})
 	api := houston.NewHoustonClient(client)
-	id := "ck1qg6whg001r08691y117hub"
+	id := "ckoixo6o501496qemiwsja1tl"
 	role := "test-role"
 	email := "andrii@test.com"
 
 	buf := new(bytes.Buffer)
 	err := UpdateRole(id, role, email, api, buf)
 	assert.NoError(t, err)
-	expected := `Role has been changed from andrii@test.com to WORKSPACE_VIEWER for user test-role`
+	expected := `Role has been changed from WORKSPACE_VIEWER to WORKSPACE_ADMIN for user test-role`
 	assert.Equal(t, buf.String(), expected)
 }
 
+func TestUpdateRoleNoAccessDeploymentOnly(t *testing.T) {
+	testUtil.InitTestConfig()
+	okResponse := `{"data":{"workspaceUpdateUserRole":"WORKSPACE_ADMIN", "workspaceUser":{"roleBindings":[{"workspace":{"id":"ckg6sfddu30911pc0n1o0e97e"},"role":"DEPLOYMENT_VIEWER"}]}}}`
+	client := testUtil.NewTestClient(func(req *http.Request) *http.Response {
+		return &http.Response{
+			StatusCode: 200,
+			Body:       ioutil.NopCloser(bytes.NewBufferString(okResponse)),
+			Header:     make(http.Header),
+		}
+	})
+	api := houston.NewHoustonClient(client)
+	id := "ckg6sfddu30911pc0n1o0e97e"
+	role := "test-role"
+	email := "andrii@test.com"
+
+	buf := new(bytes.Buffer)
+	err := UpdateRole(id, role, email, api, buf)
+	assert.Equal(t, err.Error(), "The user you are trying to change is not part of this workspace")
```

Comment:
I may be over thinking this. But is this misleading. The user has access to this deployment in this workspace at the deployment level.